src/components/EditableField/EditableField.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react'
+import propConnect from '../PropProvider/propConnect'
+import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+import { classNames } from '../../utilities/classNames'
+import { noop } from '../../utilities/other'
+import { EditableFieldProps, EditableFieldState } from './EditableField.types'
+import {
+  EditableFieldUI,
+  LabelTextUI,
+  FieldActionsUI,
+  FieldButtonUI,
+} from './styles/EditableField.css'
+import { COMPONENT_KEY } from './EditableField.utils'
+import EditableFieldInput from './EditableFieldInput'
+import { key } from '../../constants/Keys'
+
+export class EditableField extends React.PureComponent<
+  EditableFieldProps,
+  EditableFieldState
+> {
+  static className = 'c-EditableField'
+  static defaultProps = {
+    innerRef: noop,
+    initialValue: '',
+    type: 'text',
+    value: '',
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      initialValue: props.value,
+      value: props.value,
+    }
+  }
+
+  getClassName() {
+    const { className } = this.props
+    return classNames(EditableField.className, className)
+  }
+
+  getNewValue = ({ inputValue, name }) => {
+    const { value } = this.state
+    const idx = Number(name.split('_')[1])
```

Comment:
What does this do here?

Reply:
I'll make it more readable, but in a nutshell, it determines the new value for the appropriate input in case of Fields with multiple values